### DISCUSSION TITLE
GWTSwing for TeaVM

### DISCUSSION THREAD

**Discussion by pizzadox9999** [konsoletyper/teavm#702] [Show and tell]
Hi, 
I hope this is the right place to post about it ^^.
On my journey to find somehting similar to swing which runs with TeaVM I found GWTSwing from [keinhaar](https://github.com/keinhaar). He was so kind to make it platform independent and so usable with TeaVM. So for all the people who want a look a like of swing for TeaVM this should it be. Unfortunately because every swing class starting with a J was replaced with a G so an easy windowbuilder integration is not possible. This demo shows what GWTSwing is capable of: [GWTSwing Demo](http://www.exware.de/gwtswing/demo/teavm/demo.html).

With best regards pizzadox9999

**Comment by konsoletyper** [comment]
Demo link does not work for me. What is the reason to have everything with `G` prefix?

**Reply by pizzadox9999** [reply]
There are 2 main differences between Swing an GWTSwing.
  - There are no modal dialogs. Instead of that, callback objects can be used, to get notified if a dialog was closed.
  - All Classnames beginn with "G" instead of "J" (GTextField instead of JTextField). This was done to avoid license problems, because the API  
    is now different from Oracles Swing and to avoid confusion while developing with GWTSwing.

But since the OpenJDK and GWTSwing have compatible licences this shouldn't be a problem. [Licence](https://www.gnu.org/licenses/license-list.html.en)
But I'm not the author of the library. But maybe in a community effort he could be asked in a nice way for a change.

The demo link should work now.
